app/lib/package/models.dart

```diff
@@ -550,4 +550,15 @@ class PackageLinks {
       issueTrackerUrl: issueTrackerUrl,
     );
   }
+
+  /// URL to use as for linking the referenced resource relative to.
+  String get baseUrl {
+    String url = repositoryUrl ?? homepageUrl;
+    if (repositoryUrl != null &&
+        homepageUrl != null &&
+        homepageUrl.startsWith(repositoryUrl)) {
+      url = homepageUrl;
```

Comment:
Hmm... I'm not sure this is a good idea.

We should always use the `repositoryUrl` to identify the repository, **only** if there is no `repositoryUrl`, then we should try to derive one from the the `homepage`.

Adding a heuristic that tries to derive sub-folders from the `homepage` is just more confiusing. If we want that, we should define how to point to sub-folders in the `repositoryUrl` (I'm not sure this is easy to do, we would need to update docs for `pubspec.yaml`).

I think better linking to repository source is a project we could solve, but it should be a project that involves known git hash of the version too.

Reply:
IMO; if we want to do a heuristtic for this, we should do it the same way we do for `LICENSE`. Though ideally, we should implement code listing.

Reply:
I'll just remove the commit in question from this PR, and address that part later.